### ISSUE TITLE
Feature request: global configuration to automate key imports to container to remove the need to `ddev auth ssh` for each site 

### ISSUE THREAD

**Issue by mdyoung3** [ddev/ddev#4970]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Is your feature request related to a problem?

Problem: I frequently spin down docker containers and restarting them each day which necessitates running `ddev auth ssh` frequently. In fact, I try sshing with Drush enough that I often run three commands to get to sshing into the server. 

TL;DR Running into forgetting to run `ddev auth ssh` consistently because of how often I spin down Docker.  

### Describe your solution

I'm not sure the situation for local development where you would worry about importing ssh keys in your web container. Allow for configuration (global) to override default and automate importing keys into the container. 

### Describe alternatives

N/A

### Additional context

N/A

**Comment by rfay** [comment]
Just FYI, `ddev auth ssh` only needs to be done once after DDEV is started, not for each site. But I think you already understand that. `ddev poweroff` must be a fundamental part of your workflow. 

Do you actually need the keys inside the container? If not, don't worry about `ddev auth ssh`.

In some environments you may also be able to just mount the ssh socket into the container, https://stackoverflow.com/a/74033072/215713

You may also be able to just copy your keys into the container, https://stackoverflow.com/a/61485062/215713

But you could do this:

```yaml
hooks:
  post-start:
    exec-host: ddev exec ssh-add -l || ddev auth ssh
```

If you had that in your projects it would always make sure you did `ddev auth ssh`, but only if you needed it. 

Since good ssh keys have a password, the only way I actually know to automate entering the password is via `expect`, which is what DDEV automated tests do, for example, https://github.com/ddev/ddev/blob/b0ba7ff9086b4cb107c3251cba8a1134794ef90d/pkg/ddevapp/testdata/TestAcquiaPull/ddevauthssh.expect - but that has security implications if you care about your ssh keys.

**Comment by mdyoung3** [comment]
@rfay First, thanks for the speedy response. 

> Just FYI, ddev auth ssh only needs to be done once after DDEV is started

You are right. I didn't realize this before, I guess. (I like to spin down Colima after work to prevent battery drainage on laptops.)

> Do you actually need the keys inside the container?

It might not be terribly necessary down the road, but now it's handy because I run Drush inside the containers because I do both Drupal 7 and Drupal 9, with different versions of PHP. I have to use drush for accessing sites on hosting.

I was able to get the post-start hook to work. But I might look at those other options so I don't have to update a bunch of ddev configs at the project level. 

Also, in case anyone else needs it, I had to tweak the `post-start` a little to get it to work for me. Here's what I did:

```
hooks:
  post-start:
    - exec-host: ddev auth ssh
```

FYI: Docksal used to auto-import ssh keys. I'd be curious to know why Ddev doesn't do this.

**Comment by mdyoung3** [comment]
I forgot I use other tools, like Terminus and BLT, that require ssh. So, it's just kind of handy to be able to have the SSH keys already in the container.

**Comment by rfay** [comment]
DDEV doesn't import your private keys directly because it seems like a security violation. Maybe not, but copying private keys (into container) isn't really the best idea. 

I think you'll find with colima that if you aren't running any projects you won't have any battery usage. So `ddev stop -a` will stop all ddev projects but leave the ddev-ssh-agent running. If you use that workflow you'll have good battery and not have to `ddev auth ssh` so often.

I don't think your changed hook will be completely satisfying, as start on each project will require entering the passwords. That's why I suggested what I did.
